nav2_costmap_2d/test/unit/costmap_update_on_request_test.cpp

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) 2022 Samsung R&D Institute Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License. Reserved.
+
+#include <vector>
+#include <memory>
+#include <chrono>
+
+#include <gtest/gtest.h>
+
+#include <rclcpp/rclcpp.hpp>
+#include <nav_msgs/msg/occupancy_grid.hpp>
+#include <nav2_costmap_2d/cost_values.hpp>
+#include <nav2_costmap_2d/costmap_2d_ros.hpp>
+#include <nav2_util/occ_grid_values.hpp>
+
+using namespace std::chrono_literals;
+
+static constexpr double RESOLUTION = 0.05;
+static constexpr double ORIGIN_X = 0.1;
+static constexpr double ORIGIN_Y = 0.2;
+
+// Wrapper around Costmap2DROS to get access to protected member variables
+class Costmap2DROSWrapper : public nav2_costmap_2d::Costmap2DROS
+{
+public:
+  Costmap2DROSWrapper(const std::string & name)
+  : Costmap2DROS(name, "/", name, true) {}
+
+  void setup()
+  {
+    // Workaround to avoid setting base_link->map transform
+    this->set_parameter(rclcpp::Parameter("robot_base_frame", "map"));
+    this->set_parameter(rclcpp::Parameter("update_frequency", 10.0));
+
+    // Add static_layer plugin
+    std::vector<std::string> plugins_str{"static_layer"};
+    this->set_parameter(rclcpp::Parameter("plugins", plugins_str));
+    this->declare_parameter(
+      "static_layer.plugin",
+      rclcpp::ParameterValue(std::string("nav2_costmap_2d::StaticLayer")));
+
+    this->on_configure(this->get_current_state());
+    this->on_activate(this->get_current_state());
+  }
+
+  void setUpdateOnRequestParam(const bool & update_on_request)
+  {
+    this->set_parameter(rclcpp::Parameter("update_on_request", update_on_request));
+  }
+
+  void waitForMap()
+  {
+    while (!this->isCurrent()) {
```

Comment:
One thing that I can't quite figure out is if we still need to check if the layered costmap is `current`? From the static layer plugin, it is set to true after receiving a new map. I can remove this check if that is the case and just do a normal spin.